### PR TITLE
Update Microsoft.Identity.Model to 7.0.0-preview3

### PR DESCRIPTION
**Pull request by jennyf19**
# {PR title}

- [x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

## Description

Update M.IM to 7.0.0-preview3, which fully  replaces Newtonsoft.Json with System.Text.Json, and enables ASP.NET's JwtBearer auth handler will now be fully AOT compatible. See [release notes](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/releases/tag/7.0.0-preview3) for details.

Fixes #{bug number} (in this specific format)

### CODE REVIEW

File: src/Security/Authentication/WsFederation/src/WsFederationHandler.cs [hunk 1]
```diff
@@ -98,7 +98,7 @@ protected override async Task HandleChallengeAsync(AuthenticationProperties prop
         var wsFederationMessage = new WsFederationMessage()
         {
             IssuerAddress = _configuration.TokenEndpoint ?? string.Empty,
-            Wtrealm = Options.Wtrealm,
+            Wtrealm = Options.Wtrealm!,
```
Comment (eerhardt) [inline src/Security/Authentication/WsFederation/src/WsFederationHandler.cs:101]: Is this the right fix? It seems wrong.

Comment (eerhardt) [reply src/Security/Authentication/WsFederation/src/WsFederationHandler.cs:101]: It looks like the change that broke this is

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/commit/c058737642f6e04aab80fef5f15de118e7bb2adc

@brentschmaltz @westin-m @FuPingFranco - are you certain that `WsFederationMessage.Wtrealm` needs to be non-nullable?

Comment (brentschmaltz) [reply src/Security/Authentication/WsFederation/src/WsFederationHandler.cs:101]: @eerhardt When requesting a SecurityToken, wtrealm must contain a uri.
http://[docs.oasis-open.org/wsfed/federation/v1.2/os/ws-federation-1.2-spec-os.html#_Toc223175023](http://docs.oasis-open.org/wsfed/federation/v1.2/os/ws-federation-1.2-spec-os.html#_Toc223175023)

Comment (eerhardt) [reply src/Security/Authentication/WsFederation/src/WsFederationHandler.cs:101]: @halter73 @Tratcher - any thoughts here? Why is `Options.Wtrealm` annotated as `string?` when `WsFederationMessage.Wtrealm` is a non-nullable `string`. Do we have our nullable annotations incorrect in ASP.NET?

File: src/Security/Authentication/WsFederation/src/WsFederationHandler.cs [hunk 2]
```diff
@@ -195,7 +195,7 @@ protected override async Task<HandleRequestResult> HandleRemoteAuthenticateAsync
             {
                 // Extract the user state from properties and reset.
                 properties.Items.TryGetValue(WsFederationDefaults.UserstatePropertiesKey, out var userState);
-                wsFederationMessage.Wctx = userState;
+                wsFederationMessage.Wctx = userState!;
```
Comment (eerhardt) [inline src/Security/Authentication/WsFederation/src/WsFederationHandler.cs:198]: This looks like a bug in IdentityModel. `wctx` is OPTIONAL according to http://docs.oasis-open.org/wsfed/federation/v1.2/os/ws-federation-1.2-spec-os.html#_Toc223175023.

@jennyf19 @brentschmaltz - can you open an issue in https://github.com/azuread/azure-activedirectory-identitymodel-extensions-for-dotnet/ to track annotating these properties correctly for `nullable`? Then link to the bug here.

File: src/Security/Authentication/WsFederation/src/WsFederationHandler.cs (hunk 1 above)
Comment (jennyf19) [reply src/Security/Authentication/WsFederation/src/WsFederationHandler.cs:101]: @eerhardt will you fix it on your end?

Comment (eerhardt) [reply src/Security/Authentication/WsFederation/src/WsFederationHandler.cs]: I've opened https://github.com/dotnet/aspnetcore/issues/50242 to track annotating `Wtrealm` correctly, and added links to those 2 locations in the code. I believe the `Wctx` below is a bug in IdentityModel.

File: src/Security/Authentication/WsFederation/src/WsFederationHandler.cs (hunk 2 above)
Comment (jennyf19) [reply src/Security/Authentication/WsFederation/src/WsFederationHandler.cs:198]: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2240

**Review by eerhardt — APPROVED** [review]
LGTM.

@jennyf19 - Since we merged https://github.com/dotnet/source-build-externals/pull/194, we will need to take preview3 in order to unblock the build. When preview4 comes, we will update to that as well.

**Pull request merged**